app/controllers/groups/self_inscription_controller.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+#  Copyright (c) 2023, Pfadibewegung Schweiz. This file is part of
+#  hitobito and licensed under the Affero General Public License version 3
+#  or later. See the COPYING file at the top-level directory or at
+#  https://github.com/hitobito/hitobito.
+
+class Groups::SelfInscriptionController < CrudController
+
+  skip_authorization_check
+  skip_authorize_resource
+
+  before_action :redirect_to_group_if_necessary
+
+  delegate :self_registration_active?, to: :group
+
+  def create
+    @role = build_entry
+    @role.group = group
+    @role.type = group.self_registration_role_type
+    @role.person = person
+    @role.save
+    send_notification_email
+    redirect_with_message(notice: t('.role_saved'))
+  end
+
+  private
+
+  def send_notification_email   
+    return if group.self_registration_notification_email.blank?
+
+    Groups::SelfRegistrationNotificationMailer
+      .self_registration_notification(group.self_registration_notification_email,
+                                      @role).deliver_now
+  end
+
+  def return_path
+    super.presence || group_path(group, format: request.format.to_sym)
+  end
+
+  def redirect_with_message(message)
+    return redirect_to group_person_path(person.default_group_id, person), message
+  end
+
+  def redirect_to_group_if_necessary
+    if Role.where(person: person, group: group, type: group.self_registration_role_type).present?
```

Comment:
Wir sollten hier nur aktive Rollen berücksichtigen. Wenn jemand z.B. in der Vergangenheit die Rolle bereits hatte, sollte sich diese Person später dennoch erneut einschreiben können.

Reply:
Ich habe deleted_at und archived_at berücksichtigt. Stimmt das so?

Reply:
Die gelöschten Rollen (`deleted_at`) werden im default scope bereits herausgefiltert, diese müssen also nicht mehr berücksichtigt werden.